dev_tools/tests/test_nyaml2nxdl.py

```diff
@@ -1,27 +1,58 @@
-import os
+from pathlib import Path
 
+import lxml.etree as ET
 from click.testing import CliRunner
 
 from ..nyaml2nxdl import nyaml2nxdl as conv
+from ..nyaml2nxdl.nyaml2nxdl_helper import remove_namespace_from_tag
 from ..utils.nxdl_utils import find_definition_file
 
-# import subprocess
-
 
 def test_conversion():
     root = find_definition_file("NXentry")
-    # subprocess.run(["python3","-m","dev_tools.nyaml2nxdl.nyaml2nxdl","--input-file",root])
-    result = CliRunner().invoke(conv.launch_tool, ["--input-file", root])
+    result = CliRunner().invoke(conv.launch_tool, ["--input-file", str(root)])
     assert result.exit_code == 0
-    yaml = root[:-9] + "_parsed.yaml"
-    # subprocess.run(["python3","-m","dev_tools.nyaml2nxdl.nyaml2nxdl","--input-file",yaml])
-    result = CliRunner().invoke(conv.launch_tool, ["--input-file", yaml])
+    # Replace suffixes
+    yaml = root.with_suffix("").with_suffix(".yaml")  # replace .nxdl.xml
+    yaml = yaml.with_stem(yaml.stem + "_parsed")  # extend file name with _parsed
```

Comment:
```suggestion
    yaml = root.with_suffix("").with_suffix(".stem_parsed.yaml")  # replace .nxdl.xml
```
Like this?

Reply:
Do you mean,
`yaml = root.with_suffix("").with_suffix("_parsed.yaml")`

Reply:
Addressed!

Reply:
> Do you mean, `yaml = root.with_suffix("").with_suffix("_parsed.yaml")`

Sorry, neither 
`yaml = root.with_suffix("").with_suffix(".stem_parsed.yaml")`
nor 
``yaml = root.with_suffix("").with_suffix("_parsed.yaml")`
works.

In the first one, it replaces the `.nxdl.xml` as `.stem_parsed.yaml`
In the second one `_parsed.yaml` is an invalid suffix.

Reply:
It depends on what you want to do. I just created the suggestion, because there is no `with_stem` on a PosixPath, so I just combined it such that it reproduces what I thought you wanted to do with the two lines (which I thought was to change `.nxdl.xml` to `.stem_parsed.yaml` - but I now realised that you just wanted to use the `.stem` field). `_parsed.yaml` does not work because a suffix has to start with `.`, if you want to do it such the following could work:

`path.parent / Path(path.with_suffix("").stem + '_parsed.yaml')`

Reply:
It should be resolved now, we fixed it together.